aptos-move/aptos-vm/src/block_executor/mod.rs

```diff
@@ -133,6 +145,37 @@ impl BlockAptosVM {
         state_view: &S,
         concurrency_level: usize,
     ) -> Result<Vec<TransactionOutput>, VMStatus> {
+        let addrs: HashSet<AccountAddress> = transactions
+            .iter()
+            .flat_map(|txn| {
+                if let Transaction::UserTransaction(t) = txn {
+                    Some(t.sender())
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        IO_POOL.install(|| {
+            addrs.par_iter().for_each(|addr| {
+                let ap_coin = AccessPath::resource_access_path(ResourceKey::new(
+                    *addr,
+                    CoinStoreResource::struct_tag(),
+                ));
+                let ap_seq = AccessPath::resource_access_path(ResourceKey::new(
```

Comment:
@gelash Why do you access CoinStore twice here?

Reply:
ouch, thanks!